src/renderer/canvas/label.ts

```diff
@@ -58,40 +72,160 @@ export class Label {
   }
 }
 
-export const drawLabel = (context: CanvasRenderingContext2D, label: Label) => {
+export const drawLabel = (context: CanvasRenderingContext2D, label: Label, type: 'node' | 'edge') => {
```

Comment:
I don't really get why you need information about if it is a `node` or `edge`. Why does it matter?

Both can have multiple lines of text. As far as I can see, you probably wanted to handle the situation where padding is larger than the initial margin and in the node label case, the background color of the text would go over/under the node circle, right?

That's why we have this `textBaseline` property. Nodes have that set as `TOP`, and edges have it as `MIDDLE`. With that info, you have everything you need. No need to send `'node'` and/or `'edge'`.